src/odemis/dataio/catmaid.py

```diff
@@ -0,0 +1,343 @@
+
+        self._base_url = base_url
+        self._session = requests.Session()
+        self._stack_info = stack_info
+        file_extension = self._stack_info["mirrors"][0]["file_extension"]
+        self._file_extension = file_extension[1:] if file_extension.startswith(".") else file_extension
+
+    def getTile(self, x, y, zoom, depth=437):
+        """
+        Fetches one tile
+        x (0<=int): X index of the tile.
+        y (0<=int): Y index of the tile
+        zoom (0<=int): zoom level to use. The total shape of the image is shape / 2**zoom.
+            The number of tiles available in an image is ceil((shape//zoom)/tile_shape)
+        depth (0<=int): The Z index of the stack.
+        return:
+            tile (DataArray): tile containing the image data and the relevant metadata.
+        """
+        tile_width, tile_height = self.tile_shape
+        tile_url = format_tile_url(
+            tile_source_type=self._stack_info["mirrors"][0]["tile_source_type"],
+            image_base=self._stack_info["mirrors"][0]["image_base"],
+            zoom=zoom,
+            depth=depth,
+            col=x,
+            row=y,
+            file_extension=self._file_extension,
+            tile_width=tile_width,
+            tile_height=tile_height,
+        )
+        _, username, password = read_config_file(self._base_url, username=True, password=True)
+        try:
+            image = response_to_array(self._session.get(tile_url, auth=(username, password)))
+        except HTTPError as e:
+            if e.response.status_code == 401:
+                raise AuthenticationError("Authentication failed while getting tiles at {}".format(tile_url))
+            else:
+                logging.error("No tile at %s (error %s), returning blank tile", tile_url, e.response.status_code)
+                image = numpy.zeros((tile_width, tile_height))
+
+        tile = model.DataArray(image, self.metadata.copy())
+        orig_pixel_size = self.metadata.get(model.MD_PIXEL_SIZE, (1e-6, 1e-6))
+        # calculate the pixel size of the tile for the zoom level
+        tile_pixel_size = tuple(ps * 2 ** zoom for ps in orig_pixel_size)
+        tile.metadata[model.MD_PIXEL_SIZE] = tile_pixel_size
+
+        # calculate the center of the tile
+        position = get_tile_md_pos((x, y), self.tile_shape, tile, self)
+        tile.metadata[model.MD_POS] = [p for p in position]
+
+        return tile
+
+    def getData(self):
+        """Abstract method of DataArrayShadow"""
+        pass
+
+
+class AcquisitionDataCatmaid(AcquisitionData):
+    """
+    Implements AcquisitionData for Catmaid instances
+    """
+
+    def __init__(self, base_url, project_id, stack_id):
+        data = self._createDataArrayShadows(base_url, project_id, stack_id)
+        AcquisitionData.__init__(self, tuple(data))
+
+    @staticmethod
+    def _createDataArrayShadows(base_url, project_id, stack_id):
```

Comment:
As this is just 2 lines, you could just merge that function into the `__init__()`.

BTW, I realise that we could imagine mapping each stack ID of a catmaid server to a different DataArrayShadow, and return _all_ of them if the user hasn't specified one. I don't think we should implement this now, but it's good to keep in mind for later.